xls/dslx/run_routines.cc

```diff
@@ -372,7 +373,9 @@ absl::StatusOr<TestResult> ParseAndTest(std::string_view program,
     if (status.ok()) {
       std::cerr << "[            OK ]" << std::endl;
     } else {
-      handle_error(status, test_name, /*is_quickcheck=*/false);
+      auto handled = handle_error(status, test_name, /*is_quickcheck=*/false);
+      if (!options.handle_status)
```

Comment:
curious why we don't always return `status` here too?

Reply:
Always returning `status` here would change behavior of `ParseAndTest` to always fail fast. @proppy

Reply:
I see, so we want to:
a/ keep the current behavior (run all tests, count the number of failure and report individual test failure)
b/ make sure that the right error is populated (actual proc test failure, rather than type lookup error)
?

Reply:
Yes, I want to keep `a` behavior and add `b` through option.

Reply:
it seems to me that we can achieve (and test) `b/` without introducing a way to configure the failfast behavior: we just have to check for the presence of the right error message in the output (and the absence of type inference error: if we want to check for regression)

Reply:
So you would like to capture std::cerr in testing infrastructure and check it's contents?

Reply:
In example above test calls `Interpret` method https://github.com/google/xls/blob/main/xls/dslx/bytecode/bytecode_interpreter_test.cc#L400, while `procs` call `TestAndProc`.
`Interpret` directly calls Bytecode interpreter, so status returned represents exactly what has happened inside the function.
Procs due to their stateful nature require evaluation environment with eval loop. This is done by calling ParseAndTest.

Reply:
It seems that testing for:
```
status_testing::IsOkAndHolds(TestResult::kSomeFailed)
```
could allow us to test for failure without fail fast.

wdyt?

Reply:
Yes, it would give us information that tests failed, but it doesn't have reason for why each test failed.
That would lead back to ambiguity for the source of failure.

I checked and GTEST doesn't have publicly defined options to capture std:cerr or std::cout, but it has internal function to do so. I can use it to capture handlers error message and check its contents for the expected string.

Reply:
@proppy
I've removed `options.handle_status` completely in the newest commit.
I'm capturing `std::cerr` and result from ParseAndTest. Testing captured `std::cerr` catches regression.